server_installation/topics/operator/keycloak-cr.adoc

```diff
@@ -0,0 +1,232 @@
+
+* `keycloak-db-secret` - Stores properties such as the database username, password, and external address (if you connect to an external database)
+* `credentials-<CR-Name>` - Admin username and password to log into the {project_name} admin console (the `<CR-Name>` part is based on the `Keycloak` custom resource name)
+* `keycloak` - Keycloak deployment specification that is implemented as a StatefulSet with high availability support
+* `keycloak-postgresql` - Starts a PostgreSQL database installation
+* `keycloak-discovery` Service - Performs `JDBC_PING` discovery
+* `keycloak` Service - Connects to {project_name} through HTTPS (HTTP is not supported)
+* `keycloak-postgresql` Service - Connects an internal and external, if used, database instance
+* `keycloak` Route - The URL for accessing the {project_name} admin console from OpenShift
+ifeval::[{project_community}==true]
+* `keycloak` Ingress - The URL for accessing the {project_name} admin console from Kubernetes
+endif::[]
+
+.How Operator components and services interact
+image:{project_images}/operator-components.png[]
+
+==== The Keycloak custom resource
+
+The Keycloak custom resource is a YAML file that defines the parameters for installation.  This file contains three properties. 
+
+* `instances` - controls the number of instances running in high availability mode.
+* `externalAccess` - if the `enabled` is `True`, the Operator creates a route for OpenShift
+ifeval::[{project_community}==true]
+ or an Ingress for Kubernetes
+endif::[]
+ for the {project_name} cluster.
+* `externalDatabase` - applies only if you want to connect an externally hosted database. That topic is covered in the xref:_external_database[external database] section of this guide.
+
+.Example YAML file for a Keycloak custom resource
+```yaml
+apiVersion: keycloak.org/v1alpha1
+kind: Keycloak
+metadata:
+  name: keycloak
+  labels:
+    app: sso
+spec:
+  instances: 1
+  externalAccess:
+    enabled: True
+```
+
+[NOTE]
+====
+You can update the YAML file and the changes appear in the {project_name} admin console, however changes to the admin console do not update the custom resource.
+====
+
+ifeval::[{project_community}==true]
+==== Creating a Keycloak custom resource on Kubernetes
+
+On Kubernetes, you create an ingress, which is the IP address of the admin console, and find the secret, which holds the username and password for that console.
+
+.Prerequisites
+
+* You have a YAML file for this custom resource.
+
+* You have cluster administrator permissions or an equivalent level of permissions granted by an administrator.
+
+.Procedure 
+
+. Create the ingress using your YAML file. `oc create -f <filename>.yaml -n <namespace>`.  For example:
++
+[source,bash,subs=+attributes]
+----
+$ oc create -f keycloak.yaml -n keycloak
+westover.org/keycloak created
+----
++
+This command creates an ingress on Kubernetes.
```

Comment:
Shouldn't be this sentence on the line 67?

Reply:
This sentence is intended to say this is the result of the step "Create the ingress using your YAML file....."
I don't think we need it however. I will remove it.

Reply:
I see, it just confused me. So then remove it, if it is not a problem :-).